ddtrace/profiling/profiler.py

```diff
@@ -34,7 +35,7 @@ def _build_default_exporters():
 def _build_default_collectors():
     r = recorder.Recorder()
     return [
-        stack.StackCollector(r),
+        stack.StackCollector(r, tracer=ddtrace.tracer),
```

Comment:
We're probably going to need to document somewhere that if people use their own tracer instances that they'll also have to specify their own `StackCollector`. Not sure if we should build some helpers for this or not 🤔

Reply:
Agreed. The defaults works out of the box and that ought to be good enough.

For now, I expect that if people use their own tracer (or multiple tracers 😅) they know what they're doing and can build their own `Profiler` instance. We'll see if that's a common use case so that we need to improve the API at different layers.